include/nvexec/stream/common.cuh

```diff
@@ -578,7 +578,7 @@ namespace nvexec {
           operation_state_base_t<stdexec::__id<OuterReceiver>>&>;
 
     template <class Sender, class InnerReceiver, class OuterReceiver>
-      using stream_op_state_t = operation_state_t<stdexec::__id<Sender>,
+      using stream_op_state_t = operation_state_t<stdexec::__cvref_id<Sender, Sender>,
```

Comment:
The nvexec `when_all` instantiates this `stream_op_state_t` with `Sender&&`, which lead to my `static_assert` detecting `__id` being passed a reference type.

It's not clear just by looking at this code that `stream_op_state_t` is meant to be passed a possibly cvref Sender.

Reply:
I don't see a use of `stream_op_state_t` in when_all.cuh. Could you point me at the code in question?

Reply:
I got mixed up. `stream_op_state_t` is referenced by then.cuh's `connect` which passes a possible cvref qualified `Sender` to `stream_op_state_t `.


It was `exit_operation_state_t` that when_all.cuh references that also passes a cvref qualified `Sender.`

Reply:
@senior-zero, could you have a look at the changes Chris made in nvexec?

Reply:
@ericniebler, @ccotter changes look good. Stream operation should be able to accept possibly cvref sender. It might be worth adding a test case at some point.